c10/core/ScalarType.h

```diff
@@ -68,6 +71,8 @@ enum class ScalarType : int8_t {
   NumOptions
 };
 
+constexpr uint16_t NumScalarTypes = static_cast<uint16_t>(ScalarType::NumOptions);
```

Comment:
Unrelated to our current endeavor trying to fix ROCm, I think this is extra and should be removed?  You (re)define NumScalarTypes again later in `c10/util/typeid.h` as static constexpr.

Reply:
Thanks for catching this! I'll put up the change in a little while.

Reply:
Did you remove this extra statement in the final PR?